internal/controller/git/git.go

```diff
@@ -364,6 +431,23 @@ func (r *repo) GetDiffPaths() ([]string, error) {
 	return paths, nil
 }
 
+func (r *repo) GetDiffPathsForCommitID(commitID string) ([]string, error) {
+	resBytes, err := libExec.Exec(r.buildGitCommand("diff", "--name-only", commitID))
```

Comment:
I am not sure if I am misunderstanding how this func is meant to be used or if there's an actual mistake here, but isn't `git diff --name-only <commit>` exactly equivalent to `git diff --name-only <commit>..HEAD` used by `GetDiffPathsSinceCommitID`?

I presumed the intent behind this function to be returning paths modified _by_ a given commit, but I think this is returning paths modified _since_ that commit. (Same as `GetDiffPathsSinceCommitID`.)

Possible I am misunderstanding the intent.

Reply:
You are right, and it should be `git diff --name-only <commit>^!`. Great catch 🙇